src/project-roots.ts

```diff
@@ -18,52 +15,32 @@ export default class ProjectRoots {
 
   projects = new Map<string, Project>();
 
-  async initialize(workspaceRoot: string, watcher: EventEmitter) {
+  async initialize(workspaceRoot: string) {
     this.workspaceRoot = workspaceRoot;
 
-    let promise = new Promise(resolve => {
-      watcher.once('ready', resolve);
+    const roots = walkSync(workspaceRoot, {
+      directories: false,
+      globs: ['**/ember-cli-build.js'],
+      ignore: [
+        '**/.git/**',
+        '**/bower_components/**',
+        '**/dist/**',
+        '**/node_modules/**',
+        '**/tmp/**',
+      ]
```

Comment:
the caveat with this is that project discovery only runs on startup, right?

Reply:
Yes, but I see no way to improve that without file watching. Maybe we can just rerun the discovery on any new requests if x amount of time has passed since the last time. Not perfect, but I am wondering if that's such a common case.

Reply:
> but I am wondering if that's such a common case

probably not. let's do it like this for now and we can build something more sophisticated later if people start complaining... 😉